sound/soc/intel/boards/sof_da7219_max98373.c

```diff
@@ -103,6 +104,15 @@ static const struct snd_soc_dapm_route max98373_map[] = {
 	{ "Right Spk", NULL, "Right BE_OUT" },
 };
 
+/* For MAX98360A amp */
+static const struct snd_soc_dapm_widget max98360a_widgets[] = {
+	SND_SOC_DAPM_SPK("Spk", NULL),
+};
+
+static const struct snd_soc_dapm_route max98360a_map[] = {
+	{"Spk", NULL, "Speaker"},
+};
+
```

Comment:
@yongzhi1 if you do something similar to widgets and maps to what you've done with controls ie have max98360a_widgets and max98373_widgets separately, then you'd need do additional changes in the init function at all? same for maps. I think the code will look a lot cleaner then?

Reply:
> @yongzhi1 if you do something similar to widgets and maps to what you've done with controls ie have max98360a_widgets and max98373_widgets separately, then you'd need do additional changes in the init function at all? same for maps. I think the code will look a lot cleaner then?

If I understand correctly, your suggestion is to have two independent set of widgets and route for the two type of spk amp so that there is no need to have .init callback?

Reply:
precisely but only it makes sense

Reply:
Not entirely sure @yongzhi1 @ranj063 , but it could indeed be cleaner still. And wouldn't this also avoid adding the non-functional speaker control to 98373..?